.github/workflows/flow-micro-benchmarks.yml

```diff
@@ -0,0 +1,122 @@
+      - name: Setup tests dependencies
+        run: |
+          .install/test_deps/common_installations.sh ${{ steps.mode.outputs.mode }} sudo
+      - name: Install Boost
+        working-directory: .install
+        run: ./install_boost.sh ${{ env.BOOST_VERSION }} sudo
+
+      - name: Build RediSearch
+        run: make build
+            PROFILE=${{ inputs.profile_env }}
+            SHOW=1
+      - name: Prepare automation
+        run: |
+          ./deps/readies/bin/getpy3
+          pip3 install --upgrade pip
+          python3 -m pip install -r tests/benchmarks/requirements.txt
+          VERSION=1.11.1 sudo ./deps/readies/bin/getterraform
+      - name: Prepare ReJSON Module
+        run: REJSON_BRANCH=${{ inputs.rejson_branch }} ./tests/deps/setup_rejson.sh
+      
+      - name: Get Latest Baseline
+        uses: actions/download-artifact@v4
+        with: 
+          name: rust-benchmark-results-master
+          path: ${{ inputs.output_folder }}
+        continue-on-error: true # Prevent failure if no baseline exists yet
+      - name: Check if Baseline Exists
+        id: check_baseline
+        run: |
+          if [ -d ${{ inputs.output_folder }} ]; then
+            echo "baseline_exists=true" >> $GITHUB_OUTPUT
+          else
+            echo "baseline_exists=false" >> $GITHUB_OUTPUT
+          fi
+      - name: Create output directory if not existing
+        run: mkdir -p "${{ inputs.output_folder }}"    
+      - name: Run benchmark on PR with baseline from master
+        if: github.event_name == 'pull_request' && steps.check_baseline.outputs.baseline_exists == 'true'
+        run: ./memory_tracked_command.sh "cargo bench -- --baseline master" ${{ inputs.memory_limit }}
+        working-directory: ${{ inputs.benchmark_folder }}
+      - name: Run benchmark on PR without baseline
+        if: github.event_name == 'pull_request' && steps.check_baseline.outputs.baseline_exists == 'false'
+        run: ./memory_tracked_command.sh "cargo bench" ${{ inputs.memory_limit }}
+        working-directory: ${{ inputs.benchmark_folder }}
+      - name: Run benchmark on master
+        if: github.ref == 'refs/heads/master' && github.event_name == 'push' && success() 
+        run: ./memory_tracked_command.sh "cargo bench -- --save-baseline master" ${{ inputs.memory_limit }}
+        working-directory: ${{ inputs.benchmark_folder }}
+
+      - name: Upload rust baseline benchmarks for master
+        if: github.ref == 'refs/heads/master' && github.event_name == 'push' && success() 
+        uses: actions/upload-artifact@v4
+        with:
+          name: "rust-benchmark-results-master"
+          path: ${{ inputs.output_folder }}
+      - name: Upload benchmarks for PR comparison
+        if: github.event_name == 'pull_request'
+        uses: actions/upload-artifact@v4
+        with:
+          name: "rust-benchmark-results-pr-${{ github.event.pull_request.number }}"
+          path: ${{ inputs.output_folder }}  
```

Comment:
This doesn't seem to be working, looking at https://github.com/RediSearch/RediSearch/actions/runs/14612608134/job/40993542455#step:16:12